### ISSUE TITLE
no-unused-vars firing incorrectly on args inside template-tag

### ISSUE THREAD

**Issue by davidtaylorhq** [ember-cli/eslint-plugin-ember#2118]
Given this snippet:
```gjs
const array = [];

<template>
  {{#each array as |item index|}}
    {{index}}
  {{/each}}
</template>
```
Eslint will fail with

```
error  'item' is defined but never used  no-unused-vars
```

In the same file, I can do `someFunction((item, index) => console.log(index));` in regular javascript with no errors about 'item' being unused.

So it seems that the default '[`args: after-used`'](https://eslint.org/docs/latest/rules/no-unused-vars#args-after-used) behaviour is not being applied properly inside template tags.

**Comment by patricklx** [comment]
Does `_item` also get reported as unused?

**Comment by davidtaylorhq** [comment]
Underscores don't seem to make any difference out-the-box. Calling it `_item`, or `_` doesn't make any difference

If I configure an eslint `argsIgnorePattern`, then underscored variables can be ignored:

```gjs
/*eslint no-unused-vars: ["error", { "argsIgnorePattern": "^_" }]*/

const array = [];

<template>
  {{#each array as |_item index|}}
    {{index}}
  {{/each}}
</template>
```
^^ this passes ✅ 

But this `argsIgnorePattern` is not default eslint behaviour, so I imagine most projects rely on the default '[args: after-used'](https://eslint.org/docs/latest/rules/no-unused-vars#args-after-used)' behaviour.

**Comment by patricklx** [comment]
Oh, right.
Does it make a difference of its a gts or gjs file?

**Comment by davidtaylorhq** [comment]
Same issue in both gjs and gts 😓

**Comment by NullVoxPopuli** [comment]
Isn't this correct behavior tho?

**Comment by davidtaylorhq** [comment]
This fails :x:
```gjs
<template>
  {{#each array as |item index|}}
    {{index}}
  {{/each}}
</template>
```

This passes ✅ 
```js
someFunction((item, index) => console.log(index));
```

Conceptually, they are the same, so the eslint result should be the same.

Eslint [docs say](https://eslint.org/docs/latest/rules/no-unused-vars#args)

> unused positional arguments that occur before the last used argument will not be checked, but all named arguments and all positional arguments after the last used argument will be checked.
